pkg/appfile/parser.go

```diff
@@ -342,13 +343,26 @@ func (p *Parser) parseReferredObjects(ctx context.Context, af *Appfile) error {
 		if err := utils.StrictUnmarshal(comp.Properties.Raw, spec); err != nil {
 			return errors.Wrapf(err, "invalid properties for ref-objects in component %s", comp.Name)
 		}
+		if len(spec.Objects) > 0 && len(spec.URLs) > 0 {
+			return fmt.Errorf("cannot use objects and urls in ref-objects at the same time")
+		}
```

Comment:
why not, just append them together into the slice.

Reply:
Removed.